src/main/scala/io/shiftleft/js2cpg/preprocessing/TranspilationRunner.scala

```diff
@@ -11,24 +11,42 @@ import java.nio.file.{Path, StandardCopyOption}
 import scala.util.Try
 import scala.util.chaining.scalaUtilChainingOps
 
-class TranspilationRunner(projectPath: Path, tmpTranspileDir: Path, config: Config) {
+class TranspilationRunner(projectPath: Path,
+                          tmpTranspileDir: Path,
+                          config: Config,
+                          subDir: Option[Path] = None) {
 
   private val logger = LoggerFactory.getLogger(getClass)
 
-  private val transpilers: Seq[Transpiler] = Seq(
-    new TranspilerGroup(config,
-                        projectPath,
-                        Seq(
-                          new NuxtTranspiler(config, projectPath),
-                          new TypescriptTranspiler(config, projectPath),
-                          new BabelTranspiler(config, projectPath)
-                        )),
-    new VueTranspiler(config, projectPath),
-    new EjsTranspiler(config, projectPath),
-    new PugTranspiler(config, projectPath),
-  )
+  private val transpilers: Seq[Transpiler] = createTranspilers()
 
-  private def handlePrivateModules(): List[(Path, Path)] = {
+  private def createTranspilers(): Seq[Transpiler] = {
+    // We always run the following transpilers by default when not stated otherwise in the Config.
+    // This includes running them for sub-projects.
+    val baseTranspilers = TranspilerGroup(
+      config,
+      projectPath,
+      Seq(
+        new TypescriptTranspiler(config, projectPath, subDir = subDir),
+        new BabelTranspiler(config, projectPath, subDir = subDir)
+      )
+    )
+
+    // When we got no sub-project, we also run the following ones:
+    if (subDir.isEmpty) {
```

Comment:
What's the reason for this? I'm guessing the transpiler of the root project covers this for all subprojects? Can you add an explaining comment?

Reply:
Yes, correct. These three transpilers simply run only for the root project.